onert-micro/onert-micro/include/OMConfig.h

```diff
@@ -17,13 +17,86 @@
 #ifndef ONERT_MICRO_CONFIG_H
 #define ONERT_MICRO_CONFIG_H
 
+#include <stdint.h>
+#include <stdlib.h>
+
 namespace onert_micro
 {
 
+/*
+ * OMTrainOptimizer - enum to store optimizers supported by training with onert-micro
+ */
+enum OMTrainOptimizer
+{
+  SGD,
+  ADAM,
+};
+
+/*
+ * OMMetrics - enum to store metrics supported by training with onert-micro
+ */
+enum OMMetrics
+{
+  MSE_METRICS,
+  MAE_METRICS,
+  CROSS_ENTROPY_METRICS,
+  ACCURACY,
+};
+
+/*
+ * OMLoss - enum to store loss supported by training with onert-micro
+ */
+enum OMLoss
+{
+  BINARY_CROSS_ENTROPY,
+  CROSS_ENTROPY,
+  MSE,
+  MAE,
+};
+
+/*
+ * OMTrainingContext contains training specific parameters
+ * batch_size - batch size for training (Note: 1 - default one)
+ * num_of_train_layers - number of trainable last layers (Note: 0 - all layers will be trained)
+ * optimizer - optimizer which onert-micro training will be used (Note: SGD - default one)
+ * loss - loss which onert-micro training will be used (Note: CROSS_ENTROPY - default one)
+ * learning_rate - used by all optimizers
+ * beta - used by ADAM optimizer
+ * beta_squares - used by ADAM optimizer
+ * epsilon - used by ADAM optimizer
+ * num_Step - used by ADAM optimizer
+ */
+struct OMTrainingContext
+{
+  uint32_t batch_size = 1;
+  uint32_t num_of_train_layers = 0;
+  OMTrainOptimizer optimizer = SGD;
+  OMLoss loss = MSE;
+  float learning_rate = 0.f;
+  float beta = 0.9f;
+  float beta_squares = 0.9f;
+  float epsilon = 10e-8;
+  uint32_t num_step = batch_size;
```

Comment:
since this is for checkpoint, this should be 0 at start. It is better to increase this after each step internally. But, this value is not maintained inside interpreter. Thus, I try to increase this on nnfw api level like : https://github.com/Samsung/ONE/commit/89af72f45123f16f50f9efea865d9a9a9c83cfe5

Reply:
Changed to 0